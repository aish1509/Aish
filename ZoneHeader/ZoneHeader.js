import {IconButton} from '@material-ui/core';
import HomeIcon from '@material-ui/icons/Home';
import PropTypes from 'prop-types';
import React from 'react';
import {Logout, SettingsIcon} from '@math/common';
import {makeStyles} from '@material-ui/styles';

import './ZoneHeader.scss';

export const useStyles = makeStyles(theme => ({
	container: {
		display: 'flex',
		height: '60px',
		backgroundColor: theme.colors.green[100],
		justifyContent: 'space-between'
	},
	headerText: {
		fontFamily: theme.fonts.openSansBold,
		fontSize: '15px',
		padding: '15px 0 0 15px',
		color: theme.colors.blueGrey[20],
		flex: '20',
		textTransform: 'uppercase'
	},
	leftSquare: {
		display: 'flex',
		backgroundColor: theme.colors.green[40]
	},
	triangleTopleft: {
		borderTop: `60px solid ${theme.colors.green[40]}`,
		borderRight: `60px solid ${theme.colors.green[100]}`
	}
}));
const ZoneHeader = ({
	children,
	onHomeClick,
	firstName,
	lastName,
	onLogout,
	onSettingsClick,
	showSettings
}) => {
	const classes = useStyles();

	const handleOnClick = e => {
		e.preventDefault();
		onHomeClick();
	};

	return (
		<div className={classes.container}>
			<div className={classes.leftSquare}>
				<IconButton onClick={handleOnClick}>
					<HomeIcon style={{color: 'white'}} />
				</IconButton>
				<div className={classes.triangleTopleft} />
			</div>
			<div className={classes.headerText}>{children}</div>
			<Logout
				lastName={lastName}
				firstName={firstName}
				onLogout={onLogout}
			/>
			{showSettings && <SettingsIcon onSettingsClick={onSettingsClick} />}
		</div>
	);
};

ZoneHeader.defaultProps = {
	children: null,
	onLogout: () => {},
	onSettingsClick: () => {},
	showSettings: true
};
ZoneHeader.propTypes = {
	firstName: PropTypes.string.isRequired,
	lastName: PropTypes.string.isRequired,
	children: PropTypes.node,
	onLogout: PropTypes.func,
	onSettingsClick: PropTypes.func,
	showSettings: PropTypes.bool,
	onHomeClick: PropTypes.func.isRequired
};

export default ZoneHeader;
