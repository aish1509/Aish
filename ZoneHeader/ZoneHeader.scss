.zone-header {
  display: flex;
  flex-wrap: nowrap;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 25px;

  color: #fff;
  padding: 15px;
  align-content: center;
}

.zone-header--explore-zone {
  background-color: #529a44;
}

.zone-header--fast-track {
  background-color: #ff070e;
}

.zone-header--learn-zone {
  background-color: #ff8e1e;
}

.zone-header--success-zone {
  background-color: #990be2;
}

.zone-header__text {
  position: relative;
  margin-left: 5px;
  padding-top: 15px;
}
