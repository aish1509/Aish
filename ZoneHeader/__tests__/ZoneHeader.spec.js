import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import ZoneHeader, {useStyles} from '../index';
import {HeaderText} from '@math/common';

describe('<ZoneHeader />', () => {
	let wrapper = null;
	let mockOnLogout = null;
	let mockOnSettingsClick = null;
	let mockOnHomeClick = null;

	beforeEach(() => {
		mockOnLogout = jest.fn();
		mockOnSettingsClick = jest.fn();
		mockOnHomeClick = jest.fn();
		wrapper = shallow(
			<ZoneHeader
				onHomeClick={mockOnHomeClick}
				firstName="John"
				lastName="Doe"
				onLogout={mockOnLogout}
				onSettingsClick={mockOnSettingsClick}
				showSettings
			>
				<HeaderText subheaderText="Block 1: Multiplicative thinking">
					Mock Zone
				</HeaderText>
			</ZoneHeader>
		);
	});

	it('should render correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
	// it('handleOnClick', () => {
	// 	wrapper = mount(
	// 		<ZoneHeader
	// 			onHomeClick={mockOnHomeClick}
	// 			firstName="John"
	// 			lastName="Doe"
	// 			onLogout={mockOnLogout}
	// 			onSettingsClick={mockOnSettingsClick}
	// 			showSettings
	// 		>
	// 			<HeaderText subheaderText="Block 1: Multiplicative Thinking">
	// 				Mock Zone
	// 			</HeaderText>
	// 		</ZoneHeader>
	// 	);

	const mockEvent = {
		preventDefault: jest.fn()
	};
	it('handle click', () => {
		const buttonComp = wrapper.find('WithStyles(ForwardRef(IconButton))');
		expect(buttonComp).toHaveLength(1);
		buttonComp.prop('onClick')(mockEvent);
		expect(mockEvent.preventDefault).toHaveBeenCalled();
		expect(mockOnHomeClick).toHaveBeenCalled();
	});

	it('handleLogout', () => {
		wrapper.find('Logout').prop('onLogout')();
		expect(mockOnLogout).toHaveBeenCalled();
	});

	it('handleSettingsClick', () => {
		wrapper.find('SettingsIcon').prop('onSettingsClick')();
		expect(mockOnSettingsClick).toHaveBeenCalled();
	});
});
