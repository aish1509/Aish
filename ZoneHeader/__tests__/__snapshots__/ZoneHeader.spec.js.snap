// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZoneHeader /> should render correctly 1`] = `
<div
  className={
    Object {
      "backgroundColor": "#3BA570",
      "display": "flex",
      "height": "60px",
      "justifyContent": "space-between",
    }
  }
>
  <div
    className={
      Object {
        "backgroundColor": "#B1DBC6",
        "display": "flex",
      }
    }
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <HomeIcon
        style={
          Object {
            "color": "white",
          }
        }
      />
    </WithStyles(ForwardRef(IconButton))>
    <div
      className={
        Object {
          "borderRight": "60px solid #3BA570",
          "borderTop": "60px solid #B1DBC6",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "color": "#E6F0F7",
        "flex": "20",
        "fontFamily": Array [
          "OpenSans-Bold",
          "Roboto",
          "Helvetica",
          "Arial",
          "sans-serif",
        ],
        "fontSize": "15px",
        "padding": "15px 0 0 15px",
        "textTransform": "uppercase",
      }
    }
  >
    <WithStyles(HeaderText)
      subheaderText="Block 1: Multiplicative thinking"
    >
      Mock Zone
    </WithStyles(HeaderText)>
  </div>
  <Logout
    firstName="John"
    lastName="Doe"
    onLogout={[MockFunction]}
  />
  <SettingsIcon
    onSettingsClick={[MockFunction]}
  />
</div>
`;
